src/lib/y2users/autoinst_profile/user_section.rb

```diff
@@ -89,8 +89,7 @@ def self.attributes
       #   @return [String,nil] Default shell
       #
       # @!attribute user_password
-      #   @return [String,nil] User's password. If set to an exclamation mark ('!'),
-      #   a random password is generated. See #encrypted.
+      #   @return [String,nil] User's password.
```

Comment:
yeap, it is not random, but it disables login with password. Quotation from man page for shadow:
```
If the password field contains some string that is not a valid result of crypt(3), for instance ! or *, the user will not be able to use a unix password to log in (but the
           user may log in the system by other means).
```